### DISCUSSION TITLE
Filtering NEI data from industrial facilities

### DISCUSSION THREAD

**Discussion by m-jamieson** [NETL-RIC/ElectricityLCI#191] [General]
I wasn't sure whether to put this in StEWI or here. I figured since the problem is manifesting itself here, I would start here.

This is applicable to the replace_egrid=True path, but I suspect is applicable to the egrid path as well, depending on how that data was ultimately filtered to arrive at the official egrid derived plant list for 2016.

There are some generators which are located within industrial facilities. An example with a really large impact is Plant ID 60822, "Intel - Ocotillo Campus Solar," a solar PV plant. Because it is a generator in EIA923 that is reporting as _NAICS-22 Non-Cogen_, we're keeping it in the list of plants to pull inventory for. StEWI correctly finds NEI data for this plant through the facility crosswalk, and assigns those NEI emissions to our solar plant. This is an electronics manufacturing facility and happens to have a lot of SF6 emissions, which get assigned to our solar generation, resulting in ~400 kg CO2e/MWh solar (versus the usual <100 kg CO2e/MWh). The NAICS code assigned to those NEI emissions is electronics manufacturing (334413) rather than electricity generation (221114), so I think the solution would be to filter NEI (and likely other inventories) based on the NAICS code for the process/facility. This is just one example. There are a number of renewable generators that are in the same situation, and I suspect a number of oil-fired and possibly natural-gas fired generators are affected as well. 

I guess there are a couple points worth discussion:

- Could/should StEWI could/should be modified to return the NAICS codes for every emission, or if I should implement a fix here [as done with flowsa](https://github.com/USEPA/flowsa/blob/master/flowsa/data_source_scripts/stewiFBS.py#L76-L83) (thanks @bl-young!). It is my understanding that this filtering is done at a facility level and could possibly miss edge cases where a process-level emission in NEI would be within a facility categories as one thing but a process within that facility categorized as another (e.g., NOx emissions from an on-site oil-fired electricity generation). I'm also not sure that StEWI even has access to the process-level NEI data, so this may be a bit moot.
- Another possible solution would be to omit all the NAICS-22 designated plants in EIA923, but that's cutting out ~40% of generation from _Electric Utility_ + _NAICS-22_.
-  Would this better fit the envisioned library that would process electricitylci out and automatically detect/remove outliers?

In the meantime, I'll probably open up and issue and start a branch to filter the data by facility-level NAICS as done in flowsa.

**Comment by bl-young** [comment]
> I'm also not sure that StEWI even has access to the process-level NEI data, so this may be a bit moot.

StEWI can pull NEI emissions by SCC using `flowbyprocess` but we don't do any further categorization of those SCCs

**Comment by m-jamieson** [comment]
You mean you don't do anything to filter out data by SCCs? As long as I can get the code, I'm happy to implement the filtering on the electricitylci side. This seems like a reasonable solution.

**Reply by bl-young** [reply]
`stewi.getInventory('NEI', '2017', 'flowbyfacility')` is just a aggregated version of `stewi.getInventory('NEI', '2017', 'flowbyprocess')` which also includes the specific SCC code

**Reply by m-jamieson** [reply]
Well maybe confetti ball. Started looking at implementation.
electricitylci gets emissions via [stewicombo (in development branch)](https://github.com/USEPA/ElectricityLCI/blob/a38d33a19f269becb66776e71536474caf17b105/electricitylci/egrid_emissions_and_waste_by_facility.py#L25). So I think in order to do this, I would have to:

- use stewicombo to grab non-NEI data
- use this flowbyprocess call separately, filter, and then combine with the stewicombo data above
- and lastly send the combined dataset back through [aggregate_and_remove_overlap](https://github.com/USEPA/standardizedinventories/blob/6522426468dff7a4c5d432704240ea3759a4bb8b/stewicombo/overlaphandler.py#L52)

Just hesitant in case I'm missing something because I'm not really at all familiar with what goes on inside stewicombo.

**Reply by bl-young** [reply]
Yes, that is the challenge is that overlaphandler/stewicombo do not really allow for using flowbyprocess in some cases but flowbyfacility in others (because the handling of overlaps is done at the facility level).

Not an insurmountable issue, just not something we've addressed yet.
